src/test.js

```diff
@@ -529,11 +529,11 @@ Test.prototype = {
 			if ( objectType( then ) === "function" ) {
 				const resume = internalStop( test );
 				if ( config.notrycatch ) {
-					then.call( promise, function() { resume(); } );
+					then.call( promise, resume );
```

Comment:
LGTM, but worth documenting I think (perhaps at internalStop's return statement) that we're going to potentially pass that callback unrelated foreign arguments and that it must thus never accept any arguments of its own so long as that's the case.